model/src/main/scala/aqua/model/func/FuncCallable.scala

```diff
@@ -129,9 +146,24 @@ case class FuncCallable(
             Eval.now(Chain.empty)
           )
       }
-      .map { case ((_, resolvedExports), b) =>
+      .map { case ((_, resolvedExports), callableFuncBody) =>
         // If return value is affected by any of internal functions, resolve it
-        FuncOp(b) -> result.map(_.resolveWith(resolvedExports))
+        (for {
+          exp <- call.exportTo
+          res <- result
+          pair <- exp match {
+            // TODO check variable, not string
+            case Call.Export(name, StreamType(_)) if name == "-return-" =>
+              // don't create identity for `-return-` stream
+              Some(FuncOp(callableFuncBody) -> result)
+            case Call.Export(name, StreamType(_)) =>
+              // path nested function results to a stream
+              Some(FuncOps.seq(FuncOp(callableFuncBody), FuncOps.identity(res, exp)) -> result)
```

Comment:
Why not resolve the result with resolved exports?

Reply:
it shouldn't be resolved if a result is a stream. Everything should be done before